src/include/serial_inline.h

```diff
@@ -296,11 +299,16 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
     if (WT_PAGE_TRYLOCK(session, page) != 0)
         return (0);
 
-    obsolete = __wt_update_obsolete_check(session, page, upd->next, true);
+    obsolete = __wt_update_obsolete_check(session, page, upd->next, true, &upd_count);
 
     WT_PAGE_UNLOCK(session, page);
 
     __wt_free_update_list(session, &obsolete);
 
+    if (upd_count > 1000) {
```

Comment:
Instead of 1000 we can say WT_THOUSAND

Reply:
I'd suggest doing the page-eviction check and schedule in `__wt_update_obsolete_check`, rather than in its caller. There's only one caller and it simplifies the code, we don't need to pass back a count.

Reply:
Please add a comment as to why we are checking against the value 1000.

Reply:
> I'd suggest doing the page-eviction check and schedule in `__wt_update_obsolete_check`, rather than in its caller. There's only one caller and it simplifies the code, we don't need to pass back a count.

Fixed

Reply:
> Instead of 1000 we can say WT_THOUSAND

Fixed

Reply:
@tammybailey Updated the comments, Can you please review the changes now?